### PR TITLE
Add aten::fractional_max_pool2d, fractional_max_pool3d (forward and backward)

### PR DESCRIPTION
**Pull request by yucai-intel**
- [x] fractional_max_pool2d
- [x] fractional_max_pool2d_backward
- [x] fractional_max_pool3d
- [x] fractional_max_pool3d_backward

### CODE REVIEW

File: src/ATen/native/xpu/sycl/FractionalMaxPool3dKernels.cpp [hunk 1]
```diff
@@ -0,0 +1,741 @@
...
+    int plane = item.get_group()[1];
+    int batch = item.get_group()[0];
+
+    if (ourOutputPoint < gradOutputPlaneSize) {
+      int64_t outputT = ourOutputPoint / gradOutputSizeH / gradOutputSizeW;
+      int64_t outputH = ourOutputPoint / gradOutputSizeW % gradOutputSizeH;
+      int64_t outputW = ourOutputPoint % gradOutputSizeW;
+
+      int64_t index = indices[batch][plane][outputT][outputH][outputW];
+      assert(index >= 0);
```
Comment (majing921201) [inline src/ATen/native/xpu/sycl/FractionalMaxPool3dKernels.cpp:463]: SYCL_KERNEL_ASSERT used for kernel

File: src/ATen/native/xpu/sycl/FractionalMaxPool2dKernels.cpp [hunk 2]
```diff
@@ -0,0 +1,498 @@
...
+    int inputSizeH,
+    int inputSizeW,
+    int outputSizeH,
+    int outputSizeW,
+    int poolSizeH,
+    int poolSizeW,
+    const bool is_channels_last) {
+  using accscalar_t = acc_type_device<scalar_t, kXPU>;
+  auto dev_id = getDeviceIndexOfCurrentQueue();
+  int64_t max_wg_size = syclMaxWorkGroupSize(dev_id);
```
Comment (fengyuan14) [inline src/ATen/native/xpu/sycl/FractionalMaxPool2dKernels.cpp:176]: To call with a specified kernel.

Comment (fengyuan14) [reply src/ATen/native/xpu/sycl/FractionalMaxPool2dKernels.cpp:176]:
```
using FractionalMaxPool2dOutFrameKernelFunctor<scalar_t, accscalar_t> = KernelClass;
int64_t max_wg_size = syclMaxWorkGroupSize<KernelClass>(dev_id);
```

**Comment by yucai-intel** [comment]
CI res
2024-10-30T07:03:48.4594797Z test_meta_xpu.py::TestMetaXPU::test_meta_outplace_nn_functional_fractional_max_pool2d_xpu_bfloat16 PASSED [ 94%]
2024-10-30T07:03:48.4928917Z test_meta_xpu.py::TestMetaXPU::test_meta_outplace_nn_functional_fractional_max_pool2d_xpu_float16 PASSED [ 94%]
2024-10-30T07:03:48.5262203Z test_meta_xpu.py::TestMetaXPU::test_meta_outplace_nn_functional_fractional_max_pool2d_xpu_float32 PASSED [ 94%]
2024-10-30T07:03:48.5597563Z test_meta_xpu.py::TestMetaXPU::test_meta_outplace_nn_functional_fractional_max_pool2d_xpu_float64 PASSED [ 94%]
2024-10-30T07:03:48.6018217Z test_meta_xpu.py::TestMetaXPU::test_meta_outplace_nn_functional_fractional_max_pool3d_xpu_bfloat16 PASSED [ 94%]
2024-10-30T07:03:48.6442224Z test_meta_xpu.py::TestMetaXPU::test_meta_outplace_nn_functional_fractional_max_pool3d_xpu_float16 PASSED [ 94%]
2024-10-30T07:03:48.6859245Z test_meta_xpu.py::TestMetaXPU::test_meta_outplace_nn_functional_fractional_max_pool3d_xpu_float32 PASSED [ 94%]
2024-10-30T07:03:48.7279210Z test_meta_xpu.py::TestMetaXPU::test_meta_outplace_nn_functional_fractional_max_pool3d_xpu_float64 PASSED [ 94%]

2024-10-30T06:55:15.0244205Z test_meta_xpu.py::TestMetaXPU::test_dispatch_symbolic_meta_outplace_nn_functional_fractional_max_pool2d_xpu_bfloat16 PASSED [ 62%]
2024-10-30T06:55:15.0977480Z test_meta_xpu.py::TestMetaXPU::test_dispatch_symbolic_meta_outplace_nn_functional_fractional_max_pool2d_xpu_float16 PASSED [ 62%]
2024-10-30T06:55:15.1701006Z test_meta_xpu.py::TestMetaXPU::test_dispatch_symbolic_meta_outplace_nn_functional_fractional_max_pool2d_xpu_float32 PASSED [ 62%]
2024-10-30T06:55:15.2430600Z test_meta_xpu.py::TestMetaXPU::test_dispatch_symbolic_meta_outplace_nn_functional_fractional_max_pool2d_xpu_float64 PASSED [ 62%]
2024-10-30T06:55:15.3381233Z test_meta_xpu.py::TestMetaXPU::test_dispatch_symbolic_meta_outplace_nn_functional_fractional_max_pool3d_xpu_bfloat16 PASSED [ 62%]
2024-10-30T06:55:15.4329430Z test_meta_xpu.py::TestMetaXPU::test_dispatch_symbolic_meta_outplace_nn_functional_fractional_max_pool3d_xpu_float16 PASSED [ 62%]
2024-10-30T06:55:15.5267707Z test_meta_xpu.py::TestMetaXPU::test_dispatch_symbolic_meta_outplace_nn_functional_fractional_max_pool3d_xpu_float32 PASSED [ 62%]
2024-10-30T06:55:15.6210275Z test_meta_xpu.py::TestMetaXPU::test_dispatch_symbolic_meta_outplace_nn_functional_fractional_max_pool3d_xpu_float64 PASSED [ 62%]

**Comment by xytintel** [comment]
image

**Pull request merged**